### ISSUE TITLE
Vue version not detected

### ISSUE THREAD

**Issue by jonvanputten-sf** [vuejs/devtools#1694]
### Version
6.0.0

### Reproduction link
Any Vue app on localhost

### Steps to reproduce
Update to the latest version and open a page with a Vue app on Chrome

### What is expected?
To find all the Vue apps in the Vue dev tools

### What is actually happening?
No apps are showing in the dev tools and the JS console shows an error:

Cannot read properties of undefined (reading 'substr')

at registerAppJob (backend.js:949:57)
    at backend.js:937:27
    at run (backend.js:3471:16)
    at backend.js:3475:30
    at onDone (backend.js:3463:11)

### UPDATE:
Changed the reproduction link to any Vue app on localhost
I am actually using 6.0.0 instead of the latest. Chrome still lists the 6.0.0 version

**Comment by Akryum** [comment]
Your reproduction is invalid as it doesn't allow me to reproduce the issue. Please provide a valid reproduction. Thanks :pray: 
![image](https://user-images.githubusercontent.com/2798204/153235256-b727293b-36ad-44de-87e6-54b696418688.png)

**Comment by muster-mark** [comment]
I am also having a similar issue. Afraid I can't provide a reproduction as it would essentially be the entired project. It seems this issue is widespread.

<img width="1264" alt="Screenshot 2022-02-09 at 17 02 55" src="https://user-images.githubusercontent.com/12281979/153251733-52e68ead-7889-4e47-bb08-0242ddf92010.png">

**Comment by Akryum** [comment]
@jwbats read the docs about installing the previous version.

**Comment by Akryum** [comment]
@jonvanputten-sf are you sure you are on version 6.0.3?

**Comment by Akryum** [comment]
@muster-mark isn't your issue this one? #1687

**Comment by muster-mark** [comment]
> @muster-mark isn't your issue this one? #1687

Ah yes, looks like it. Thank you.

**Comment by muster-mark** [comment]
@Akryum But after updating to 6.0.3, I get the same error: 
```
Uncaught (in promise) TypeError: options.version is undefined
    registerAppJob moz-extension://9845a722-2fae-4531-81d7-264c48a95821/build/backend.js:949
```

 That is on Firefox.

Meanwhile on Chrome, which is running 6.0.0, I get the same error as in this Issue:

```
backend.js:949 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'substr')
    at registerAppJob (backend.js:949:57)
```

**Comment by jwbats** [comment]
I don't get the 'options.version' error. But I am getting the 'substr' error:

```
backend.js:949 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'substr')
    at registerAppJob (backend.js:949:57)
    at backend.js:937:27
    at run (backend.js:3471:16)
    at backend.js:3475:30
    at onDone (backend.js:3463:11)
```

**Comment by Akryum** [comment]
@jwbats Because you are on Chrome and not on Firefox so the error message is different

**Comment by jwbats** [comment]
I've installed the [legacy vue devtools](https://chrome.google.com/webstore/detail/vuejs-devtools/iaajmlceplecbljialhhkmedjlpdblhp).

It solves my problem for the time being.

**Comment by jwbats** [comment]
The [official page](https://chrome.google.com/webstore/detail/vuejs-devtools/nhdogjmejiglipccpnnnanhbledajbpd?hl=en) still lists 6.0.0 from February 8th.

Where's the docs for installing specific versions?

**Comment by Akryum** [comment]
@jwbats It takes Google a few days to review the updates. They are currently reviewing 6.0.2. Firefox has [all the versions](https://addons.mozilla.org/en-US/firefox/addon/vue-js-devtools/versions/) (they take 5 minutes to publish a new version).

**Comment by lackneets** [comment]
> I don't get the 'options.version' error. But I am getting the 'substr' error:
> 
> ```
> backend.js:949 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'substr')
>     at registerAppJob (backend.js:949:57)
>     at backend.js:937:27
>     at run (backend.js:3471:16)
>     at backend.js:3475:30
>     at onDone (backend.js:3463:11)
> ```

I figure out that `Vue.version` is undefined while Vue is a VueComponent instance instead of Vue ( [Source](https://github.com/vuejs/vue/blob/612fb89547711cacb030a3893a0065b785802860/dist/vue.esm.js#L5176) )

It seems to be resolved since 6.0.5 (Fix [Source](https://github.com/vuejs/devtools/commit/dc0f5de11c73c953140fade6aa97bf278d6315b3))

Before Chrome extension updates, we could simply patch like this:

```js
import Vue from 'vue/dist/vue.esm.js'

// Prevent Vue-devtools v6.0 can't detect apps
Vue.extend({}).constructor.prototype.version = Vue.version
```

**Comment by jonvanputten-sf** [comment]
Hi @Akryum , I just noticed I have version 6.0.0 installed. I assumed that I would have the latest version when installing the dev tools. Talking about the chrome extansion btw, it still shows 6.0.0.

**Comment by jonvanputten-sf** [comment]
> @jwbats It takes Google a few days to review the updates. They are currently reviewing 6.0.2. Firefox has [all the versions](https://addons.mozilla.org/en-US/firefox/addon/vue-js-devtools/versions/) (they take 5 minutes to publish a new version).

I guess Ill wait for the reviews to finish in order to get the latest version

**Comment by Akryum** [comment]
You can also build the devtools from sources: https://devtools.vuejs.org/guide/contributing.html

**Comment by muster-mark** [comment]
@Akryum It works on 6.0.5 (I skipped 6.0.4). Thanks.

**Comment by jwbats** [comment]
Glad to hear it. I'll get 6.0.5 for Chrome as soon as it's available.

**Comment by chubbsMcfly** [comment]
Our team is also experiencing this issue, (on vue 2.x)

**Comment by jwbats** [comment]
I also had it on vue 2.

**Comment by Ranmus** [comment]
Same here, version undefined on Vue 2.x. Our team lost posibility to debug app.

**Comment by Akryum** [comment]
Should be fixed since 6.0.4.
